File: 03-lukema/learn/leetcode/src/main/java/com/learn/tree/ListNode.java
package com.learn.tree;


import lombok.ToString;


@ToString
public class ListNode {

    int val;
    ListNode next;

    ListNode() {
    }

    ListNode(int val) {
        this.val = val;
    }

    ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    public static ListNode toListNode(int[] nums) {
        if (nums == null || nums.length == 0) {
            return null;
        }

        ListNode head = new ListNode(nums[0]);
        ListNode curr = head;

        for (int i = 1; i < nums.length; i++) {
            curr.next = new ListNode(nums[i]);
            curr = curr.next;
        }

        return head;
    }
}

File: 03-lukema/learn/leetcode/src/py/lc/ListNode.py
"""
ListNode
"""

from __future__ import annotations
from typing import List


class ListNode:
    """
    ListNode
    """
    def __init__(self, val=0, next_node=None):
        self.val = val
        self.next = next_node

    @staticmethod
    def toListNode(nums: List[int]) -> (ListNode | None):
        "toListNode"
        if nums is None or len(nums) == 0:
            return None

        root = ListNode(nums[0])

        last = root

        for idx in range(1, len(nums)):
            curr = ListNode(nums[idx])
            last.next = curr
            last = curr

        return root

    @staticmethod
    def print(ln: ListNode) -> None:
        "print"
        vals = []
        while ln is not None:
            vals.append(ln.val)
            ln = ln.next

        print(vals)
